addons/sale_project/models/project.py

```diff
@@ -94,7 +94,8 @@ def _compute_has_any_so_to_invoice(self):
     def _compute_sale_order_count(self):
         sale_order_items_per_project_id = self._fetch_sale_order_items_per_project_id({'project.task': [('state', 'in', self.env['project.task'].OPEN_STATES)]})
         for project in self:
-            sale_order_lines = sale_order_items_per_project_id.get(project.id, self.env['sale.order.line'])
+            # Use sudo to avoid AccessErrors when the SOLs belong to different companies.
+            sale_order_lines = sale_order_items_per_project_id.get(project.id, self.env['sale.order.line']).sudo()
             project.sale_order_line_count = len(sale_order_lines)
             project.sale_order_count = len(sale_order_lines.order_id)
```

Comment:
I would prefer putting the sudo where we really need it, in that case, we just need it when we want to get `order_id`, the ORM will do the prefetch and so check the access at the same time. 🙂 

```suggestion
            sale_order_lines = sale_order_items_per_project_id.get(project.id, self.env['sale.order.line'])
            project.sale_order_line_count = len(sale_order_lines)
            # Use sudo to avoid AccessErrors when the SOLs belong to different companies.
            project.sale_order_count = len(sale_order_lines.sudo().order_id)
```